### PR TITLE
Move Effects into //lib/effect

### PR DESCRIPTION
**Pull request by wmmiii**
This is the start of normalizing our repo structure to follow PIO
standards.

### CODE REVIEW

File: software/generic/LedManager.cpp [hunk 1]
```diff
@@ -2,21 +2,22 @@
...
-#include "LightningEffect.hpp"
-#include "PoliceEffect.hpp"
-#include "RainbowBumpsEffect.hpp"
-#include "RainbowEffect.hpp"
-#include "RorschachEffect.hpp"
-#include "SimpleBlinkEffect.hpp"
-#include "SparkEffect.hpp"
-#include "StopLightEffect.hpp"
-#include "SwingingLights.hpp"
+#include "../../lib/effect/ColorCycleEffect.hpp"
```
Comment (ademuri) [inline software/generic/LedManager.cpp]: Do these need to include the `../..`? Can they be `effect/ColorCycleEffect.hpp`? If not, should we make a single `.hpp` file which includes all of the effects? This is a common strategy, I haven't really thought about whether it's good.

File: software/generic/test/FakeLedManager.hpp [hunk 2]
```diff
@@ -25,7 +25,7 @@ class FakeLedManager : public LedManager {
   void WriteOutLeds() override;
 
  private:
-  // Note: this looks like a pointer, but is actually an array of size num_leds
+  // Note: this looks like a pointer, but is actually an array of size numLeds
```
Comment (ademuri) [inline software/generic/test/FakeLedManager.hpp]: s/numLeds/num_leds

File: software/generic/test/FakeLedManager.hpp [hunk 3]
```diff
@@ -4,13 +4,13 @@
...
-#include "../../generic/Effect.hpp"
-#include "../../generic/LedManager.hpp"
+#include "../../lib/effect/Effect.hpp"
+#include "../LedManager.hpp"
 
 class FakeLedManager : public LedManager {
  public:
   FakeLedManager(const DeviceDescription *device,
-                 RadioStateMachine *state_machine);
+                 RadioStateMachine *stateMachine);
```
Comment (ademuri) [inline software/generic/test/FakeLedManager.hpp]: s/stateMachine/state_machine

**Review by ademuri — APPROVED** [review]

File: software/generic/LedManager.cpp (hunk 1 above)
Comment (wmmiii) [reply software/generic/LedManager.cpp]: Yeah, they need the prefix. I don't think that's the worst idea as long as its only the effects.

**Pull request merged**